File: WEEK5/bj1_11657.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
let lineIndex = 0;

const [N, M] = input[lineIndex++].split(' ').map(Number);
const edges = [];
const dist = new Array(N + 1).fill(Infinity);

for (let i = 0; i < M; i++) {
    const [A, B, C] = input[lineIndex++].split(' ').map(Number);
    edges.push({ start: A, end: B, time: C });
}

dist[1] = 0;

for (let i = 1; i < N; i++) {
    for (let j = 0; j < M; j++) {
        const edge = edges[j];
        if (dist[edge.start] !== Infinity && dist[edge.end] > dist[edge.start] + edge.time) {
            dist[edge.end] = dist[edge.start] + edge.time;
        }
    }
}

let hasNegativeCycle = false;
for (let j = 0; j < M; j++) {
    const edge = edges[j];
    if (dist[edge.start] !== Infinity && dist[edge.end] > dist[edge.start] + edge.time) {
        hasNegativeCycle = true;
        break;
    }
}

const sb = [];
if (hasNegativeCycle) {
    sb.push("-1");
} else {
    for (let i = 2; i <= N; i++) {
        if (dist[i] === Infinity) {
            sb.push("-1");
        } else {
            sb.push(dist[i]);
        }
    }
}
console.log(sb.join('\n'));

File: WEEK5/bj1_11657.py
import sys

input = sys.stdin.readline
INF = float('inf')

class Edge:
    def __init__(self, start, end, time):
        self.start = start
        self.end = end
        self.time = time

N, M = map(int, input().split())
edges = []
dist = [INF] * (N + 1)

for _ in range(M):
    A, B, C = map(int, input().split())
    edges.append(Edge(A, B, C))

dist[1] = 0

for i in range(1, N):
    for j in range(M):
        edge = edges[j]
        if dist[edge.start] != INF and dist[edge.end] > dist[edge.start] + edge.time:
            dist[edge.end] = dist[edge.start] + edge.time

hasNegativeCycle = False
for j in range(M):
    edge = edges[j]
    if dist[edge.start] != INF and dist[edge.end] > dist[edge.start] + edge.time:
        hasNegativeCycle = True
        break

if hasNegativeCycle:
    print("-1")
else:
    for i in range(2, N + 1):
        if dist[i] == INF:
            print("-1")
        else:
            print(dist[i])